### ISSUE TITLE
Missing or insufficient permissions after upgrade to Firebase Web SDK v9 

### ISSUE THREAD

**Issue by todorpr** [firebase/firebase-js-sdk#5575]
### Describe your environment

* Operating System version: MacOS Mojave
* Browser version: Chrome, Version 94.0.4606.71 (Official Build) (x86_64)
* Firebase SDK version: 9.1.1
* Firebase Product: auth

### Describe the problem

I upgraded firebase SDK from v8.2.3 to v9.1.1 following the official documentation https://firebase.google.com/docs/web/modular-upgrade. When run the application I get:

```js
Uncaught (in promise) FirebaseError: Missing or insufficient permissions.
    at index.esm2017.js:4332
    at es (index.esm2017.js:4330)
    at Zr.onMessage (index.esm2017.js:11272)
    at index.esm2017.js:11226
    at index.esm2017.js:11249
    at index.esm2017.js:15092
    at index.esm2017.js:15125
```

#### Relevant Code:

Here are the diff changes I made:

![image](https://user-images.githubusercontent.com/3307316/136036352-1bc3ea6b-1b8d-43f8-9528-e73862cf1892.png)

And eventually when I call `firestore` I changed this:

```javascript
  const firestore = firebase.firestore()
  const messagesRef = firestore
    .collection(config.firebaseRootCollection)
    .doc(uid)
    .collection(config.firebaseMessagesCollection)
```

to this:
```js
  const db = getFirestore(firebaseApp)
  const q = query(collection(db, config.firebaseRootCollection))
  const messagesRef = await getDocs(q)
  console.log(messagesRef)
```

**Comment by hsubox76** [comment]
I notice 2 things:
1) you're not passing firebaseApp into `getFirebaseAuth()` (which seems to be the renamed `getAuth()` from firebase/auth). You shouldn't have to, it should fall back to the default app, but just wondering if anything changes if you do.
2) In the firestore change you've changed the query to get all docs in the firebaseRootCollection instead of docs in the firebaseMessagesCollection of one doc in the firebaseRootCollection (with docID `uid`). Do you have different security rules for the different levels (such as one that only lets the user get the firebaseRootCollection doc that matches their uid perhaps)?

**Comment by todorpr** [comment]
Thanks for your input, @hsubox76!
1. Just tried to pass `firebaseApp` to `getFirebaseAuth()` - no difference.
2. Not sure what should be the right query with the new syntax, so that it equals the old one. Here are the rules:

```
rules_version = '2';
service cloud.firestore {
  match /databases/{database}/documents {
    match /rooms/{userId}/{messagesSubCollection}/{messages} {
        allow read: if userId == request.auth.uid;
    }
  }
}
```

**Comment by hsubox76** [comment]
Yeah, it looks like you're trying to look at every document in `rooms` which you don't have access to. It looks like we don't have updated docs showing how to get subcollections? I'll look into that.

The easiest way to get the docs from the subcollection would be to put the whole path to the subcollection:
```
const messagesRef = collection(db, `${config.firebaseRootCollection}/${uid}/${config.firebaseMessagesCollection}`);
const messagesSnapshot = await getDocs(messagesRef);
```
You can also supply each path segment as an argument to `collection()`:
```
const messagesRef = collection(db, config.firebaseRootCollection, uid, config.firebaseMessagesCollection);
const messagesSnapshot = await getDocs(messagesRef);
```

Note that getDocs returns a promise with a snapshot, not a reference. So when it returns above, you can do something like:
```
messagesSnapshot.forEach(doc => console.log(doc.data());
```
You don't need to wrap the collection reference in a `query()` unless you want to sort or filter.

**Comment by todorpr** [comment]
That is amazing, @hsubox76 - the error disappeared this way :)

Actually in the original code it does:

```js
  const firestore = firebase.firestore()
  const messagesRef = firestore
    .collection(config.firebaseRootCollection)
    .doc(uid)
    .collection(config.firebaseMessagesCollection)

  unsubscribeFn = messagesRef
    .orderBy('eventTimestamp')
    .onSnapshot(onSnapshotData, onError)
```

but now with the new approach `messagesSnapshot` doesn't have `orderBy/onSnapshot` properties. How is it supposed to be?
